### PR TITLE
Feature/exclude resources

### LINKED ISSUE #116: Add support for "opt out" to target resources
So that resource owners can explicitly exclude resources (eg: EC2, database) from power management:
By Tag?

### PR DESCRIPTION
**Pull request by alutman-innablr**
https://github.com/Innablr/revolver/issues/116

Adds a setting to specify filters for excluding resources from processing. Uses same logic as the central powercycle matcher filters.

e.g. this would ignore all ec2 resources in us-east-1
```yaml
    excludeResources:
      - type: "ec2"
      - region: "us-east-1"
```

### CODE REVIEW

**Comment by lyricnz** [comment]
If the central+powercycle both look at the "Schedule" tag, maybe it would be simpler to just have an "ignore" value in the existing Tag?  Or maybe this already exists?

**Comment by lyricnz** [comment]
I feel like this is a general "requirement", not specific to central config or existing.   A magic value, like "24x7" which means "leave me alone - don't start me, don't stop me".

**Comment by alutman-innablr** [comment]
It's more generic than central, just uses filters in the central PR. You can specify whatever tag/value you want to ignore a resource or use some more complicated logic.

**Comment by lyricnz** [comment]
I'd rather not add a new Tag/Value.  Prefer to have a special value in the existing "Schedule" Tag. 
Doesn't even the central-powercycle plugin look at target-resource Tags for an override?

**Comment by alutman-innablr** [comment]
It's just an example in the config and not a new tag/value. You can specify any existing tag or value, either in-house or by revolver.

**Comment by alutman-innablr** [comment]
Draft until central-config is merged, annoying to rebase this each time

File: test/bundle-test-config.yaml [hunk 1]
```diff
@@ -7,6 +7,10 @@ defaults:
     organizationRoleName: none
     revolverRoleName: none
     localResourcesFile: bundle-test-resources.json
+    audit:
+      console:
+    excludeResources:
+      - tag: 'Schedule|0x7'
```
Comment (lyricnz) [inline test/bundle-test-config.yaml]: Can you change this to `Schedule|ignore`?

0x7 has a real meaning: resources will be assertively shutdown, not just ignored.

File: revolver-config-example.yaml [hunk 2]
```diff
@@ -13,6 +13,11 @@ defaults:
       csv:
         file: "audit.csv"
         append: true
+    excludeResources:
+      - tag:
+          name: "Schedule"
+          contains: "ignore"
+      - region: "us-east-1"
```
Comment (lyricnz) [inline revolver-config-example.yaml:27]: Maybe add a comment saying "exclude resources with Tag:Schedule=ignore AND in us-east-1"
(assuming it's AND not OR)

Comment (alutman-innablr) [reply revolver-config-example.yaml:27]: Readme for this references the Filter section which describes what happens; If an array, AND.

**Review by simon-anz — APPROVED** [review]
LGTM. Tested with following filter, which performed as expected.

```
    excludeResources:
      or:
        - and:
          - type: ec2
          - tag: ["Name|junk-vm-3-days"]
        - and:
          - type: rdsInstance
          - id: mysql-on
```

Will need to figure out how to split this into multiple files with https://github.com/Innablr/revolver/issues/104

**Pull request merged**